op-dispute-mon/mon/extract/caller.go

```diff
@@ -17,33 +17,52 @@ import (
 
 const metricsLabel = "game_caller_creator"
 
-type GameCallerMetrics interface {
+type CallerMetrics interface {
 	caching.Metrics
 	contractMetrics.ContractMetricer
 }
+
 type GameCaller interface {
 	GetGameMetadata(context.Context, rpcblock.Block) (common.Hash, uint64, common.Hash, types.GameStatus, uint64, error)
 	GetAllClaims(context.Context, rpcblock.Block) ([]faultTypes.Claim, error)
 	BondCaller
 	BalanceCaller
 }
 
-type GameCallerCreator struct {
-	m      GameCallerMetrics
-	cache  *caching.LRUCache[common.Address, *contracts.FaultDisputeGameContract]
+type WethCaller interface {
```

Comment:
We probably should follow the same pattern here as we do for `BalanceCaller` and just include this in the `GameCaller` interface.  The `DelayedWETH` contract is encapsulated by the `FaultDisputeGame` one so the `FaultDisputeGameContract.GetBalance` call actually gets the ETH balance from the `DelayedWETH` contract.

Then we simplify most of this code as we can just keep using a single contract.

Reply:
Yeah looking at the other PRs I think following the same pattern as `BalanceCaller` will work really well since we're always requesting withdrawals for a specific game anyway. And it should just mean the stacked PRs just change from calling `wethCaller` to `gameCaller` and are otherwise unchanged.